go-controller/pkg/ovn/controller/services/service_tracker.go

```diff
@@ -49,7 +51,7 @@ func (st *serviceTracker) updateService(name, namespace, virtualIP string, proto
 
 	// check if the service already exists and create a new entry if it does not
 	vips, ok := st.virtualIPByService[serviceNN]
-	if !ok {
+	if !ok || vips == nil {
```

Comment:
can you have a service without vips?

Reply:
This was leftover from a first attempt at a fix. The way the code currently is I don't think we will hit it, but it doesn't hurt to guard against this as you could have st.virtualIPByService[serviceNN] = nil, which would then cause ok to be true, but would panic on Insert.